src/node_jsvmapi_types.h

```diff
@@ -51,6 +50,7 @@ enum napi_valuetype {
   napi_symbol,
   napi_object,
   napi_function,
+  napi_external,
```

Comment:
This is a weird object type since it's not quite an ES6 type- should this be something like napi_object_with_external_data to further clarify its relation to an ES6 type? Or should this be a modifier bit so that any es6 type can be created with external data (i.e napi_valuetypes have high-order modifier bits to control creation of the object- this could enable other things like creating arbitrary objects with special napi hooks etc)